### PR TITLE
Write three layered web application

### PR DESCRIPTION
**Pull request by ivantcacenco**

### CODE REVIEW

File: src/main/java/com/endava/internship/webapp/controller/DepartmentController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/departments")
+public class DepartmentController {
+
+    @Autowired
```
Comment (alexandrumalina) [inline src/main/java/com/endava/internship/webapp/controller/DepartmentController.java:20]: autowired is not a good style
use lombok and private final fields

File: src/main/java/com/endava/internship/webapp/controller/DepartmentController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    private DepartmentService departmentService;
+
+    @GetMapping
+    public List<Department> getAllDepartments() {
+        return departmentService.getAllDepartments();
+    }
+
+    @GetMapping("/{department_id}")
+    public Department getDepartmentById(@PathVariable("department_id") int id) {
```
Comment (alexandrumalina) [inline src/main/java/com/endava/internship/webapp/controller/DepartmentController.java:29]: I would recommend using wrapper classes in order to escape default value setting (:

File: src/main/java/com/endava/internship/webapp/controller/DepartmentController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        return departmentService.getDepartmentById(id);
+    }
+
+    @PostMapping
```
Comment (alexandrumalina) [inline src/main/java/com/endava/internship/webapp/controller/DepartmentController.java:33]:
```suggestion
    @PostMapping
        public Department saveDepartment(@RequestBody **final** Department department) {
        return departmentService.saveDepartment(department);
    }
```
please add it everywhere

File: src/main/java/com/endava/internship/webapp/model/Employee.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Table(name = "employees")
+public class Employee {
+
+    @Id
+    @Column
+    @GeneratedValue(generator = "employees_seq")
+    @SequenceGenerator(name = "employees_seq", sequenceName = "EMPLOYEES_SEQ", allocationSize = 1)
+    private int employee_id;
```
Comment (alexandrumalina) [inline src/main/java/com/endava/internship/webapp/model/Employee.java:29]: maybe use java-noted field names here?

File: src/main/resources/db/migration/V20220422182000__create_table_departments.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE departments
+(
+    department_id   NUMBER(4) PRIMARY KEY,
+    department_name VARCHAR2(30) NOT NULL,
+    location        VARCHAR2(30) NOT NULL,
+    CONSTRAINT dept_name_ne CHECK (LENGTH(department_name) > 0),
+    CONSTRAINT loc_ne       CHECK (LENGTH(location) > 0)
+);
```
Comment (alexandrumalina) [inline src/main/resources/db/migration/V20220422182000__create_table_departments.sql:8]: nl

File: src/main/java/com/endava/internship/webapp/controller/DepartmentController.java (hunk 1 above)
Comment (ivantcacenco) [reply src/main/java/com/endava/internship/webapp/controller/DepartmentController.java]: Done

File: src/main/java/com/endava/internship/webapp/controller/DepartmentController.java (hunk 2 above)
Comment (ivantcacenco) [reply src/main/java/com/endava/internship/webapp/controller/DepartmentController.java]: Done

File: src/main/java/com/endava/internship/webapp/controller/DepartmentController.java (hunk 3 above)
Comment (ivantcacenco) [reply src/main/java/com/endava/internship/webapp/controller/DepartmentController.java:33]: Done

File: src/main/resources/db/migration/V20220422182000__create_table_departments.sql (hunk 5 above)
Comment (ivantcacenco) [reply src/main/resources/db/migration/V20220422182000__create_table_departments.sql]: Done

**Review by alexandrumalina — APPROVED** [review]

**Pull request merged**